asapdiscovery-ml/asapdiscovery/ml/utils.py

```diff
@@ -28,6 +33,12 @@ def build_dataset(
         Which model to create. Current options are ["2d", "schnet", "e3nn"]
     exp_fn : str
         JSON file giving experimental results
+    xtal_pat : str, default=r"[^/]+$"
+        Regex pattern for extracting crystal structure ID from filename. Defaults to
+        taking the entire file basename (on *nix platforms)
+    compound_pat : str, default=r"[^/]+$"
```

Comment:
for these kind of regex strings in docstrings, would you be able to give a brief explanation of what they do? I'm very bad at reading them. ChatGPT thinks this one's:
> matches any string of characters, where all letters and numbers are at least one character long.

Reply:
I updated the description, hopefully it's a bit more clear now